code/src/layouts/modals/auth/components/sign-up-form/SignupForm.test.tsx

```diff
@@ -0,0 +1,99 @@
+import { fireEvent, screen, waitFor, act } from "@testing-library/react";
+import SignupForm from "@/layouts/modals/auth/components/sign-up-form/SignupForm";
+import useSignUp from "@/hooks/use-sign-up/useSignUp";
+import { renderWithProviders } from "@/utils/test-utils";
+
+jest.mock("@/hooks/use-sign-up/useSignUp", () => ({
+  __esModule: true,
+  default: jest.fn()
+}));
+
+const mockSignUp = jest.fn();
+(useSignUp as jest.Mock).mockReturnValue([mockSignUp, { isLoading: false }]);
+
+const mockFormValues = {
+  email: "test@example.com",
+  password: "Helloworld123!",
+  firstName: "Eugene",
+  lastName: "Snow"
+};
+
+describe("SignupForm", () => {
+  beforeEach(() => {
+    renderWithProviders(<SignupForm />);
+  });
+
+  test("renders input fields", () => {
+    const firstNameField = screen.getByLabelText(/signUp.firstname.field/i);
+    expect(firstNameField).toBeInTheDocument();
+  });
+
+  test("handles input changes and form submission", async () => {
+    const emailInput = screen.getByLabelText(/signUp.email.field/i);
+    const passwordInput = screen.getByLabelText(/signUp.password.field/i);
+    const confirmPasswordInput = screen.getByLabelText(
+      /signUp.confirmpassword.field/i
+    );
+    const firstNameInput = screen.getByLabelText(/signUp.firstname.field/i);
+    const lastNameInput = screen.getByLabelText(/signUp.lastname.field/i);
+
+    await act(async () => {
+      fireEvent.change(emailInput, { target: { value: mockFormValues.email } });
+      fireEvent.change(passwordInput, {
+        target: { value: mockFormValues.password }
+      });
+      fireEvent.change(confirmPasswordInput, {
+        target: { value: mockFormValues.password }
+      });
+      fireEvent.change(firstNameInput, {
+        target: { value: mockFormValues.firstName }
+      });
+      fireEvent.change(lastNameInput, {
+        target: { value: mockFormValues.lastName }
+      });
```

Comment:
Did you think about creating test util `typeIntoInput`?

Example:
```jsx
import { act, fireEvent } from "@testing-library/react";

export default async function typeIntoInput(
  input: Parameters<typeof fireEvent.change>[0],
  value: string | number
) {
  await act(async () => {
    fireEvent.change(input, { target: { value } });
  });
}
```

Reply:
Nice idea. Added